kolibri/core/tasks/worker.py

```diff
@@ -170,9 +182,17 @@ def start_next_job(self, job):
             job_id=job.job_id,
         )
 
-        # assign the futures to a dict, mapping them to a job
-        self.job_future_mapping[future] = job
-        self.future_job_mapping[job.job_id] = future
+        # Acquire locks before modifying dictionaries
+        with self.job_future_mapping_lock:
+            with self.future_job_mapping_lock:
+                # Check if the job ID already exists in the future_job_mapping dictionary
+                if job.job_id in self.future_job_mapping:
+                    logger.error(f"Job ID {job.job_id} is already in future_job_mapping.")
```

Comment:
Heads up that we can't use f-strings as we currently still have support for Python 2.7 - will need to use older string formatting methods instead!

Reply:
reformatted and pushed. thank you!